### PR TITLE
update auto-updater

### PR DESCRIPTION
**Pull request by ppot**
This update the updater process for Hyper using a single entry point

### CODE REVIEW

**Comment by chabou** [comment]
Good catch, it is better to check for update availability only once but notification should occur in all opened windows, not only the last focused one.

File: app/auto-updater.js [hunk 1]
```diff
@@ -1,50 +1,45 @@
...
-
-function init() {
-  autoUpdater.on('error', (err, msg) => {
-    console.error('Error fetching updates', msg + ' (' + err.stack + ')');
-  });
-
-  autoUpdater.setFeedURL(`${FEED_URL}/${version}`);
-
-  setTimeout(() => {
-    autoUpdater.checkForUpdates();
```
Comment (chabou) [inline app/auto-updater.js]: It is better to keep this first call.
We can't wait 30mn after start to notify user that an update is available. If user uses Hyper 15mn per day, he'll never see updates.

File: app/auto-updater.js [hunk 2]
```diff
@@ -1,50 +1,45 @@
...
   }
+};
 
-  const {rpc} = win;
-
-  const onupdate = (ev, releaseNotes, releaseName) => {
-    rpc.emit('update available', {releaseNotes, releaseName});
-  };
-
-  autoUpdater.on('update-downloaded', onupdate);
```
Comment (chabou) [inline app/auto-updater.js]: We need this callback.

File: app/auto-updater.js [hunk 3]
```diff
@@ -1,50 +1,51 @@
...
-    autoUpdater.checkForUpdates();
-  }, ms('30m'));
-
-  isInit = true;
-}
 
-module.exports = function (win) {
-  if (!isInit) {
-    init();
+    autoUpdater.on('checking-for-update', () => {
```
Comment (chabou) [inline app/auto-updater.js]: What is this event?
Never triggered 😞

**Comment by ppot** [comment]
Closing since can't work on this unless I have an Apple developper ID and can code sign.

**Pull request closed without merging**